geomstats/geometry/discrete_surfaces.py

```diff
@@ -0,0 +1,687 @@
+            v_areas = self.space.vertex_areas(base_point)
+            if self.a2 > 0:
+                Del = self.space.get_laplacian(base_point)
+                norm += self.a2 * gs.sum(
+                    gs.einsum("bi,bi->b", Del(h), Del(k)) / v_areas
+                )
+            if self.a0 > 0:
+                norm += self.a0 * gs.sum(v_areas * gs.einsum("bi,bi->b", h, k))
+        if self.a1 > 0 or self.b1 > 0 or self.c1 > 0 or self.b1 > 0:
+            alpha = self.space.surface_one_forms(base_point)
+            g = gs.matmul(gs.transpose(alpha, axes=(0, 2, 1)), alpha)
+            areas = gs.sqrt(gs.linalg.det(g))
+            n = self.space.normals(base_point)
+            if self.c1 > 0:
+                dn1 = self.space.normals(point1) - n
+                dn2 = self.space.normals(point2) - n
+                norm += self.c1 * gs.sum(gs.einsum("bi,bi->b", dn1, dn2) * areas)
+            if self.d1 > 0 or self.b1 > 0 or self.a1 > 0:
+                ginv = gs.linalg.inv(g)
+                alpha1 = self.space.surface_one_forms(point1)
+                alpha2 = self.space.surface_one_forms(point2)
+                if self.d1 > 0:
+                    xi1 = alpha1 - alpha
+                    xi1_0 = gs.matmul(
+                        gs.matmul(alpha, ginv),
+                        gs.matmul(gs.transpose(xi1, (0, 2, 1)), alpha)
+                        - gs.matmul(gs.transpose(alpha, axes=(1, 2)), xi1),
+                    )
+                    xi2 = alpha2 - alpha
+                    xi2_0 = gs.matmul(
+                        gs.matmul(alpha, ginv),
+                        gs.matmul(gs.transpose(xi2, (0, 2, 1)), alpha)
+                        - gs.matmul(gs.transpose(alpha, axes=(1, 2)), xi2),
+                    )
+                    norm += self.d1 * gs.sum(
+                        gs.einsum(
+                            "bii->b",
+                            gs.matmul(
+                                xi1_0,
+                                gs.matmul(ginv, gs.transpose(xi2_0, axes=(0, 2, 1))),
+                            ),
+                        )
+                        * areas
+                    )
+                if self.b1 > 0 or self.a1 > 0:
+                    dg1 = gs.matmul(gs.transpose(alpha1, axes=(0, 2, 1)), alpha1) - g
+                    dg2 = gs.matmul(gs.transpose(alpha2, axes=(0, 2, 1)), alpha2) - g
+                    ginvdg1 = gs.matmul(ginv, dg1)
+                    ginvdg2 = gs.matmul(ginv, dg2)
+                    norm += self.a1 * gs.sum(
+                        gs.einsum("bii->b", gs.matmul(ginvdg1, ginvdg2)) * areas
+                    )
+                    norm += self.b1 * gs.sum(
+                        gs.einsum("bii->b", ginvdg1)
+                        * gs.einsum("bii->b", ginvdg2)
+                        * areas
+                    )
+        return norm
+
+    def squared_norm(self, vector, base_point):
+        """Squared norm of a tangent vector at a base point.
```

Comment:
Docstring one-liner needs to start with verb at imperatif form
--> Compute squared norm...